docs: update fleet/agent pipeline docs (#90659)
elastic/elasticsearch: docs/reference/ingest.asciidoc

[[pipelines-for-fleet-elastic-agent]]
=== Pipelines for {fleet} and {agent}

{agent} integrations ship with default ingest pipelines that preprocess and enrich data before indexing.
{fleet-guide}/index.html[{fleet}] applies these pipelines using <<index-templates,index
templates>> that include <<set-default-pipeline,pipeline index settings>>. {es}
matches these templates to your {fleet} data streams based on the
{fleet-guide}/data-streams.html#data-streams-naming-scheme[stream's naming
scheme].

Each default integration pipeline calls a nonexistent, unversioned `@custom` ingest pipeline.

> If left untouched, this pipeline call has no effect on your data.
> However, this pipeline call can be used to create custom pipelines for integrations that persist through upgrades.

Reviewer: ```suggestion
If unaltered, this pipeline call has no effect on your data. However, you can modify this call to
create custom pipelines for integrations that persist across upgrades.
```

> See {fleet-guide}/data-streams-pipeline-tutorial.html[Tutorial: Transform data with custom ingest pipelines] to learn more.

Reviewer: ```suggestion
Refer to {fleet-guide}/data-streams-pipeline-tutorial.html[Tutorial: Transform data with custom ingest pipelines] to learn more.
```

> {fleet} doesn't provide a default ingest pipeline for the **Custom logs** integration.
> You can specify a pipeline for this integration in one of two ways: an

Reviewer: ```suggestion
{fleet} doesn't provide a default ingest pipeline for the **Custom logs** integration,
but you can specify a pipeline for this integration using an 
```
